crates/ibc/src/applications/transfer/amount.rs

```diff
@@ -28,6 +29,54 @@ impl parity_scale_codec::WrapperTypeDecode for Amount {
 #[cfg(feature = "parity-scale-codec")]
 impl parity_scale_codec::WrapperTypeEncode for Amount {}
 
+#[cfg(feature = "borsh")]
+impl borsh::BorshSerialize for Amount {
+    fn serialize<W: borsh::maybestd::io::Write>(
+        &self,
+        writer: &mut W,
+    ) -> borsh::maybestd::io::Result<()> {
+        // Note: a "word" is 8 bytes (i.e. a u64)
+        let words = self.as_slice();
+        let bytes: Vec<u8> = words.iter().flat_map(|word| word.to_be_bytes()).collect();
+
+        writer.write_all(&bytes)
+    }
+}
+#[cfg(feature = "borsh")]
+impl borsh::BorshDeserialize for Amount {
+    fn deserialize_reader<R: borsh::maybestd::io::Read>(
+        reader: &mut R,
+    ) -> borsh::maybestd::io::Result<Self> {
+        const NUM_BYTES_IN_U64: usize = 8;
+        const NUM_WORDS_IN_U256: usize = 4;
+
+        let mut buf = [0; 32];
+        let bytes_read = reader.read(&mut buf)?;
+        if bytes_read != 32 {
+            return Err(borsh::maybestd::io::Error::new(
+                borsh::maybestd::io::ErrorKind::InvalidInput,
+                format!("Expected to read 32 bytes, read {bytes_read}"),
+            ));
+        }
```

Comment:
Since the `buf`` is of fixed 32 size, would this error ever occur?

Reply:
Yes, if somehow we read less than 32 bytes